### PR TITLE
Electrical, AC

### PR DESCRIPTION
**Pull request by timmathews**
New addition to the spec: Electrical AC parameters. This covers PGNs 65001 - 65030.

### CODE REVIEW

File: schemas/groups/electrical_ac.json [hunk 1]
```diff
@@ -0,0 +1,998 @@
...
+                    "type": "object",
+                    "title": "Bus #1, Phase A",
+                    "properties": {
+                        "timestamp": {
+                            "$ref": "../definitions.json#/definitions/timestamp"
+                        },
+                        "source": {
+                            "$ref": "../definitions.json#/definitions/source"
+                        },
+                        "lineLineVoltage": {
```
Comment (tkurki) [inline schemas/groups/electrical_ac.json]: Is the 'lineLine' correct? Sounds strange.

File: schemas/groups/electrical_ac.json [hunk 2]
```diff
@@ -0,0 +1,998 @@
...
+                            "units": "VA",
+                            "minimum": -2000000000,
+                            "maximum": 2211081215,
+                            "example": 10500
+                        }
+                    }
+                }
+            }
+        },
+        "generator": {
```
Comment (tkurki) [inline schemas/groups/electrical_ac.json]: What if there are multiple generators? Like batteries, engines.

**Comment by tkurki** [comment]
First of all I'm no AC expert and don't know what all these readings are about.

I like the electrical.ac prefix.

There is a *lot* of repetition at least between the phases judging from a cursory glance at this. Should the schema be more DRY or are there notable differences that prohibit using one definition?

**Comment by tkurki** [comment]
I think it would be a good idea to come up with a realistic, full blown example document. That could be used as a runnable test against the schema as well as part of the documentation, as schema by itself makes a bit confusing reading. See https://github.com/SignalK/specification/blob/511260c9de4b8ccb8a390bc2d54cdf516bf38df1/test/multiple_values.js.  

If you can come up with a sample json document I can patch in the test.

**Comment by canboat** [comment]
Aw, another instance where the NMEA has made matters worse in trying to be precise.

In general AC electricity is generated in 3 phases. Three coils in the generator, each generating one phase, conducted using three 'line' conductors and one 'neutral' conductor. Depending on the setup, you either connect single phase consumers between phase and neutral (="Y" or "star") or between two phases (="Delta"). The voltage between two 'line' conductors is colloquially know as the _line voltage_ and the voltage between neutral and one line or phase as the _phase voltage_.

Side note: common usage in Europe is to generate phase-to-phase voltage of 400V and neutral-to-phase 230V, and connect consumers between 1 phase and neutral. The problem starts when people then say that the 'line voltage is 230 V' which is not correct -- the line voltage is 400 V and the phase voltage is 230 V.

I suggest we either use a strictly correct wording, that everybody understands, `neutralToPhaseVoltage` and `phaseToPhaseVoltage` or the more practical terms `lineVoltage` and `phaseVoltage`.

**Comment by timmathews** [comment]
We can't actually blame NMEA for this one, at least not directly. They chose to adopt the J1939-75 standard for gensets and AC power, so these PGNs come from that.

lineLineVoltage is their terminology, I agree that it could probably be simpler if we used phasePhaseVoltage, phaseNeutralVoltage, etc. (Not sure the To is necessary). But, at least in North America, most non-commercial power isn't 3-phase, it's split-phase (which is a bad term meaning that we get our power from center-tapped transformers fed from a single leg of a three-phase generator). So, typical household power is 120/240V with two hot (Line in common terminology here) and a Neutral. And we call each of them a line and both together a line. Lazy yanks.

While we muse over names I will try to simplify the schema somewhat by using definitions to remove the repetition.

File: schemas/groups/electrical_ac.json (hunk 1 above)
Comment (fabdrol) [inline schemas/groups/electrical_ac.json]: @timmathews maybe the `To` would be good to include, I agree with @tkurki that `lineLine` sounds weird.

**Comment by fabdrol** [comment]
@timmathews this can be merged, shall I?

**Comment by timmathews** [comment]
:+1: Go for merge.

This will give us parity with NMEA for AC info. We can address multiple generators, etc. later.

**Pull request merged**